doc-files/release-notes-24.md

```diff
@@ -0,0 +1,221 @@
+java --upgrade-module-path=/path/to/javafx-sdk-24/lib
+```
+
+NOTE: The above will fail if you run your application with JDK 23 or earlier. JDK 24 is recommended when running JavaFX 24, but if you choose to run JavaFX 24 with an earlier JDK, use the `--module-path` option instead.
+
+#### Applications using `jlink` to create a custom Java runtime image:
+
+When creating your custom Java runtime image, put the JavaFX jmods on the module path ahead of the JDK jmods. For example:
+
+```
+jlink --output jdk-with-javafx \
+    --module-path /path/to/javafx-jmods-24:/path/to/jdk-24/jmods \
+    --add-modules ALL-MODULE-PATH
+```
+
+NOTE: The above will fail if you create a custom image using JDK 23 or earlier. JDK 24 is recommended with JavaFX 24, but if you choose to run JavaFX 24 with an earlier JDK, put the JDK jmods ahead of the JavaFX jmods on the module path (that is, reverse the order of `javafx-jmods-24` and `jdk-24/jmods`).
+
+See [JDK-8337280](https://bugs.openjdk.org/browse/JDK-8337280) for more information.
+
+### Pluggable Image Loading via javax.imageio
+
+JavaFX 24 supports the Java Image I/O API, allowing applications to use third-party image loaders in addition to the built-in image loaders. This includes the ability to use variable-density image loaders for formats like SVG. When an image is loaded using a variable-density image loader, JavaFX rasterizes the image with the screen's DPI scaling.
+
+Applications that want to use this feature can use existing open-source Image I/O extension libraries, or register a custom Image I/O service provider instance with the IIORegistry class. Refer to the Java Image I/O documentation for more information.
+
+See [JDK-8306707](https://bugs.openjdk.org/browse/JDK-8306707) for more information.
+
+### ScrollPane Consumes Navigation Keys Only When It Has Direct Focus
+
+`ScrollPane` now only responds to key events when it is the active focus owner. This ensures that custom controls and other UI elements work correctly inside a `ScrollPane`, providing a more consistent and intuitive navigation experience.
+
+Applications that prefer the previous behavior, where `ScrollPane` always reacts to arrow keys and other navigational inputs, can manually restore it by adding an event handler. See [this note](notes/24/JDK-8340852-ScrollPane.md) for an example of how to do this.
+
+See [JDK-8340852](https://bugs.openjdk.org/browse/JDK-8340852) for more information.
+
+## Removed Features and Options
+
+### JavaFX No Longer Supports Running With a Security Manager
+
+The Java Security Manager has been permanently disabled in JDK 24 via [JEP 486](https://openjdk.org/jeps/486).
+
+Likewise, as of JavaFX 24, it is no longer possible to run JavaFX applications with a security manager enabled. This is true even if you run your application on an older JDK that still supports the security manager.
+
+The following exception will be thrown when the JavaFX runtime is initialized if the Security Manager is enabled:
```

Comment:
I think this would flow better as:

```suggestion
The following exception will be thrown when the JavaFX runtime is initialized with the Security Manager enabled:
```